service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStep.java

```diff
@@ -0,0 +1,187 @@
+
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureVmResource resource;
+  private final LandingZoneApiDispatch landingZoneApiDispatch;
+  private final AuthenticatedUserRequest userRequest;
+
+  public EnableVmLoggingStep(
+      AzureConfiguration azureConfig,
+      CrlService crlService,
+      ControlledAzureVmResource resource,
+      LandingZoneApiDispatch landingZoneApiDispatch,
+      AuthenticatedUserRequest userRequest) {
+    this.azureConfig = azureConfig;
+    this.crlService = crlService;
+    this.resource = resource;
+    this.landingZoneApiDispatch = landingZoneApiDispatch;
+    this.userRequest = userRequest;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    getDataCollectionRuleFromLandingZone()
+        .ifPresent(
+            dcr -> {
+              final String vmId =
+                  context.getWorkingMap().get(AzureVmHelper.WORKING_MAP_VM_ID, String.class);
+
+              addMonitorAgentToVm(context, vmId);
+
+              createDataCollectionRuleAssociation(context, dcr, vmId);
+            });
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Traced
+  private void createDataCollectionRuleAssociation(
+      FlightContext context, ApiAzureLandingZoneDeployedResource dcr, String vmId) {
+    getMonitorManager(context)
+        .diagnosticSettings()
+        .manager()
+        .serviceClient()
+        .getDataCollectionRuleAssociations()
+        .createWithResponse(
+            vmId,
+            resource.getVmName(),
+            new DataCollectionRuleAssociationProxyOnlyResourceInner()
+                .withDataCollectionRuleId(dcr.getResourceId()),
+            Context.NONE);
+  }
+
+  @Traced
+  private void addMonitorAgentToVm(FlightContext context, String vmId) {
+    var virtualMachine = getComputeManager(context).virtualMachines().getById(vmId);
+
+    var extension =
+        virtualMachine
+            .update()
+            .defineNewExtension("AzureMonitorLinuxAgent")
+            .withPublisher("Microsoft.Azure.Monitor")
+            .withType("AzureMonitorLinuxAgent")
+            .withVersion("1.22")
+            .withMinorVersionAutoUpgrade();
+
+    // use the pet identity if one is defined, otherwise a system identity will be used
+    Optional.ofNullable(context.getWorkingMap().get(AzureVmHelper.WORKING_MAP_PET_ID, String.class))
+        .ifPresentOrElse(
+            (petId ->
+                extension.withPublicSetting(
+                    "authentication",
+                    Map.of(
+                        "managedIdentity",
+                        Map.of("identifier-name", "mi_res_id", "identifier-value", petId)))),
+            (() -> enableSystemAssignedIdentity(virtualMachine)));
+
+    extension.attach().apply();
+  }
+
+  private void enableSystemAssignedIdentity(VirtualMachine virtualMachine) {
```

Comment:
With this change are we now saying that the default is system assigned identity for all VMs?  What is the case/scenario where they won't have a user assigned identity?

Reply:
In reality I don't know a use case when there is no user assigned identity. The code supports it however.